packages/astro/src/core/ignore-files.ts

```diff
@@ -0,0 +1,28 @@
+import fs from 'node:fs';
+import { joinPaths } from './path.js';
+
+/**
+ * use .gitignore file to ignore files
+ * @param root astro root path
+ */
+export async function ignoreFiles(root: string) {
+	const res: string[] = [];
+	const gitIgnoreFile = joinPaths(root, '.gitignore');
+	const text = await fs.promises
+		.readFile(gitIgnoreFile, 'utf-8')
+		.then((data) => data.toString())
+		.catch(() => '');
+	
+	if (text.length > 0) {
+		const lines = text.split('\n');
+		for (const line of lines) {
+			if (!line.startsWith('#')) {
+				const value = line.trim();
+				if (value.length) {
+					res.push(`**/${value}/**`);
```

Comment:
Doesn't this change the semantics of the paths inside `.gitignore`? If I have a `file.js` inside `.gitignore`, this will become `**/file.js/**`, which doesn't seem to be a valid pattern. I think we should add the paths as they are without tempering them.

Reply:
fixed

https://github.com/guobin211/astro/blob/bugfix/watch-ignore-files/packages/astro/src/core/get-gitignore-paths.ts